src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/transformer/ApplicationsTransformerTest.kt

```diff
@@ -581,4 +584,50 @@ class ApplicationsTransformerTest {
     assertThat(result.createdByUserId).isEqualTo(application.getCreatedByUserId())
     assertThat(result.status).isEqualTo(ApplicationStatus.rejected)
   }
+
+  @Test
+  fun `transformJpaToApiSummary transforms an in progress Temporary Accommodation application correctly`() {
+    val application = object : DomainTemporaryAccommodationApplicationSummary {
+      override fun getRiskRatings() = objectMapper.writeValueAsString(PersonRisksFactory().produce())
+      override fun getId() = UUID.fromString("2f838a8c-dffc-48a3-9536-f0e95985e809")
+      override fun getCrn() = randomStringMultiCaseWithNumbers(6)
+      override fun getCreatedByUserId() = UUID.fromString("836a9460-b177-433a-a0d9-262509092c9f")
+      override fun getCreatedAt() = Timestamp(Instant.parse("2023-04-19T13:25:00+01:00").toEpochMilli())
+      override fun getSubmittedAt() = null
+      override fun getLatestAssessmentSubmittedAt() = null
+      override fun getLatestAssessmentDecision() = null
+      override fun getLatestAssessmentHasClarificationNotesWithoutResponse() = false
+      override fun getHasBooking() = false
+    }
+
+    val result = applicationsTransformer.transformDomainToApiSummary(application, mockk(), mockk()) as TemporaryAccommodationApplicationSummary
+
+    assertThat(result.id).isEqualTo(application.getId())
+    assertThat(result.createdByUserId).isEqualTo(application.getCreatedByUserId())
+    assertThat(result.status).isEqualTo(ApplicationStatus.inProgress)
+    assertThat(result.risks).isNotNull
```

Comment:
Is this enough of a test? It looks like the `applicationTransformer` is using an `objectMapper` to load risk data for a person, if present it's also sent to the `risksTransformer` to transform it again into `PersonRisks`. In our test we are only asserting something is present so presumably trusting the typing to make sure whatever it is, is of type `PersonRisks`. Should we be mocking these other classes and testing the interactions?

Reply:
The main `TemporaryAccommodationApplicationEntity` class defines its `riskRatings` property to be of type `PersonRisks`. While theoretically the row in the database could contain arbitrary JSON data, I think in practice the only way this would happen is with a manual `INSERT` SQL query.

There doesn't seem to be a huge amount of testing around the risk data in general (there are no unit tests for `PersonRisksTransformer`, for example). It'd be a good idea to tighten up some of these tests, but this PR is a blocker for the frontend work so I haven't invested a huge amount of time in putting the additional infrastructure in place.